### ISSUE TITLE
Pipelines sometimes fails

### ISSUE THREAD

**Issue by saibotsivad** [saibotsivad/bpr-npm-audit#1]
After having used this for a while, I found that the command fails about 1 out of every ~40 times with two errors.

1: Sometimes it fails to push the report to Bitbucket.

```
npx bpr-npm-audit
+ npx bpr-npm-audit
npx: installed 1 in 1.191s
Could not push report to Bitbucket. 500 <!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <title>Something went wrong &mdash; Bitbucket</title>
  <link rel="stylesheet" href="https://d301sr5gafysq2.cloudfront.net/0b0967e95eb4/css/entry/bare.css">
</head>
//snip
```

2: Sometimes it fails to connect to the npm registry.

```
TODO: find example of error log 😅
```

In both these cases, rerunning the Pipeline resolves the issue.

I think a reasonable solution would be: if the process fails for any reason (thrown error internally) simply rerun it after a second or two of delay.

**Comment by saibotsivad** [comment]
Update: I've been using this in a handful of repos, with many hundreds of Pipeline runs, and have not seen this error occur again. Might have been early Pipelines instability issues?

I'm going to close as not-reproducible, but if anyone else experiences this problem please let me know!

**Comment by thaithcock** [comment]
Hey there, we've started updating our pipelines to node 18 and they are running into the connection issue you described, and don't seem to resolve after retrying a number of times. It doesn't seem to happen for node 16, though, which is odd.

Any thoughts? Have you run into this again since you closed the issue?

**Comment by saibotsivad** [comment]
That is very curious, but it's not exactly related. Some fellow engineers helped me track down the problem, I'll push out an update momentarily!

**Comment by saibotsivad** [comment]
@thaithcock fixed in [v1.4.3](https://github.com/saibotsivad/bpr-npm-audit/releases/tag/v1.4.3) which is live on npm now. Thanks for letting me know!
